providers/cloudformation/resources/template.go

```diff
@@ -0,0 +1,229 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package resources
+
+import (
+	"github.com/aws-cloudformation/rain/cft"
+	"go.mondoo.com/cnquery/v11/llx"
+	"go.mondoo.com/cnquery/v11/providers-sdk/v1/plugin"
+	"go.mondoo.com/cnquery/v11/providers-sdk/v1/util/convert"
+	"go.mondoo.com/cnquery/v11/providers/cloudformation/connection"
+	"go.mondoo.com/cnquery/v11/types"
+	"go.mondoo.com/ranger-rpc/codes"
+	"go.mondoo.com/ranger-rpc/status"
+)
+
+func initCloudformationTemplate(runtime *plugin.Runtime, args map[string]*llx.RawData) (map[string]*llx.RawData, plugin.Resource, error) {
+	if len(args) > 0 {
+		return args, nil, nil
+	}
+
+	conn := runtime.Connection.(*connection.CloudformationConnection)
+	template := conn.CftTemplate()
+
+	args["version"] = llx.StringData("")
+	args["description"] = llx.StringData("")
+	args["transform"] = llx.NilData
+
+	version, err := template.GetSection(cft.AWSTemplateFormatVersion)
+	if err == nil {
+		args["version"] = llx.StringData(version.Value)
+	}
+
+	desc, err := template.GetSection(cft.Description)
+	if err == nil {
+		args["description"] = llx.StringData(desc.Value)
+	}
+
+	transform, err := template.GetSection(cft.Transform)
+	if err == nil && len(transform.Content) > 0 {
+		var entries []string
+		for _, entry := range transform.Content {
+			entries = append(entries, entry.Value)
+		}
+		args["transform"] = llx.ArrayData(convert.SliceAnyToInterface(entries), types.String)
+	}
+
+	return args, nil, nil
+}
+
+func (r *mqlCloudformationTemplate) id() (string, error) {
+	return "cloudformation", nil
+}
+
+func (r *mqlCloudformationTemplate) extractDict(section cft.Section) (map[string]interface{}, error) {
+	conn := r.MqlRuntime.Connection.(*connection.CloudformationConnection)
+	template := conn.CftTemplate()
+	_, parameters, err := gatherMapValue(template.Node.Content[0], string(section))
```

Comment:
is there a guarantee content will have >0 elems? else this will panic

Reply:
Great catch, will add a safeguard